### PR TITLE
Add platform checks for various sys values

### PR DESCRIPTION
**Pull request by CraftSpider**

### CODE REVIEW

File: stdlib/3/sys.pyi [hunk 1]
```diff
@@ -190,7 +191,8 @@ if sys.version_info >= (3, 7):
     __breakpointhook__: Any  # contains the original value of breakpointhook
     def breakpointhook(*args: Any, **kwargs: Any) -> Any: ...
 
-def setdlopenflags(__flags: int) -> None: ...  # Linux only
+if sys.platform == "linux":
```
Comment (hauntsaninja) [inline stdlib/3/sys.pyi:194]: Docs say this is available on Unix, and can confirm it's available on my Mac

**Comment by CraftSpider** [comment]
I think the failure is coming from this line of the mypy tests: https://github.com/python/mypy/blob/5cfb0561e624b79365ac5333cbde7ff500249740/test-data/stdlib-samples/3.2/test/support.py#L476

The test isn't specifically for platform, so the usage of a platform-specific function is invalid. Does anyone with more knowledge of mypy test suite know?

**Comment by hauntsaninja** [comment]
Ugh, that's annoying. I think I might have actually tried to make this change before.

mypy seems pretty receptive to removing stdlib samples that are showing their age, see https://github.com/python/mypy/pull/8838#issuecomment-630070909. I think test.support might be tricky to remove though, in which case maybe just change the check to be against `sys.platform` (or type-ignore that line or unconditionally define those constants).

**Comment by JelleZijlstra** [comment]
Yes, probably easiest to make the mypy code use `sys.platform` instead as @hauntsaninja said. I can merge a mypy PR doing that.

**Comment by CraftSpider** [comment]
Opened a PR to mypy as https://github.com/python/mypy/pull/8916

**Pull request merged**